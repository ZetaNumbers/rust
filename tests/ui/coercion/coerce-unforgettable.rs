//@ build-fail

#![feature(unforgettable_types)]

#[derive(Debug, Default)]
struct Unforgettable {
    _unforgettable: std::marker::PhantomUnforgettable,
}

fn main() {
    let _: &dyn Send = &Unforgettable::default();
    //~^ ERROR `PhantomUnforgettable` cannot be forgotten
}
