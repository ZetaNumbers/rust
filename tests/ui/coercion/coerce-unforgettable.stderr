error[E0277]: `PhantomUnforgettable` cannot be forgotten
  --> $DIR/coerce-unforgettable.rs:11:24
   |
LL |     let _: &dyn Send = &Unforgettable::default();
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^ `PhantomUnforgettable` cannot be forgotten
   |
   = help: within `Unforgettable`, the trait `Forget` is not implemented for `PhantomUnforgettable`, which is required by `Unforgettable: Forget`
note: required because it appears within the type `Unforgettable`
  --> $DIR/coerce-unforgettable.rs:6:8
   |
LL | struct Unforgettable {
   |        ^^^^^^^^^^^^^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
